env.go

```diff
@@ -140,14 +159,49 @@ func getEnvCredentials(hostname string) (username, password string, found bool)
 	return
 }
 
-func getEcrToken() (username, password string, err error) {
-	ctx := context.TODO()
-	cfg, err := config.LoadDefaultConfig(ctx)
+// getEcrToken retrieves ECR authentication credentials (username and password) for the specified AWS account and hostname.
+// It uses AWS SDK configuration with a custom retry mechanism (10 attempts max, 5 second max backoff)
+// and a custom credentials provider that checks for account-specific environment variables.
+// The ECR authorization token is retrieved with a 30 second timeout, decoded from base64,
+// and split into username:password format. Debug mode will log token expiration time.
+//
+// Parameters:
+//
+//	hostname: The ECR repository hostname
+//	account: The AWS account ID
+//	region: The AWS region for the ECR repository
+//
+// Returns:
+//
+//	username: The decoded username (typically "AWS")
+//	password: The decoded password token
+//	err: Any error encountered during the process
+func getEcrToken(hostname, account, region string) (username, password string, err error) {
```

Comment:
It's disproportionately bugging me that you're redundantly passing both normalised and de-normalised arguments. Can we reduce this to just:
```suggestion
func getEcrToken(account, region string) (username, password string, err error) {
```
? …And just use a format string to reconstruct authoritative hostname if desired for logs?

Reply:
Done!